bloom-housing/bloom
# feat: pwdless create account flow

# Pull Request Template

## Issue Overview

This PR addresses #3818


## Description

Backend updates
* `requestSingleUseCode` moved out of the auth module and into the user module
* `confirmAndSetCredentials` added when a user logs in w single use codes that will confirm the account if it is not already
* If a user is creating a public account, toggles whether or not we send the confirmation email or a single use code based on the jurisdiction the user is in

## How Can This Be Tested/Reviewed?

Designs are [here](https://www.figma.com/file/2LSD4B4yBIxAUGbZQCVOCv/Passwordless?type=design&node-id=1607-68638&mode=design&t=8ux2qjfuaqqx2xMJ-0) 

Ensure `SHOW_PWDLESS=TRUE` on the public site, and in `devSeeding` on the backend, set `allowSingleUseCodeLogin` to true and then run `yarn setup:dev`. Relying fully on the backend var would require changes to how we're statically fetching the jurisdiction on the public site that I don't think we have space for in this pass.

* If you create an account while pwdless is toggled on, you should be taken to the verify page. When you verify the account you should be logged in and confirmed (you could check confirmation in the DB or sign out and then try to sign in with your password). If you resend the code the copy in the message changes.
* Mandated accounts should work w the create account flow (be signed out, apply to a listing, create an account, ensure you are redirected back to the app)

Known differences
* The error state for an invalid code is the bigger box version, and will not highlight the form field
* The blue messages are not closeable and do include the email address
* DOB helper text is below the field due to changes that would need to happen in UIC

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
